### PR TITLE
Fix Yarn & NPM shim issues on Windows

### LINKED ISSUE #187: Yarn-installed Spago doesn't work on Windows
If I install Spago via Yarn (doesn't matter whether locally or globally), it seems that the shell wrappers are out of order somehow:

```
> yarn add spago
...
> npx spago
/bin/bash: C:\my_project\node_modules\.bin\..\spago\spago.exe: No such file or directory
```

At the same time, I cannot use Stack-installed Spago, because then it doesn't pick up my local Node environment, and ends up using my globally installed `purs`, which is of a different version:

```
>  spago build
spago.exe: Oh noes! It looks like the PureScript version installed on your system is
outdated for the package-set you're using.

installed version:   0.12.0
package-set version: 0.12.5

Please upgrade your `purs` version.
```

So the bottom line is: Spago is pretty much unusable to me. :-/

### PR DESCRIPTION
**Pull request by stkb**
### Description of the change

(Hopefully) a fix for the shim issues on Windows described in https://github.com/spacchetti/spago/issues/187#issuecomment-494029142. Namely:
- Shims created by Yarn are broken because they try to run spago via bash.
- Shims created by NPM for bash don't work because they reference an undefined `$basedir` variable

To fix the NPM issue in particular, instead of having the package try to shim to the spago binary directly, a "2nd-level shim" nodejs script is used. This script then runs the spago binary. Since this shim doesn't
get changed during the package install process, Yarn doesn't get confused either. The script is only used on Windows.

So the result is this:
- The binary declared in the package's `package.json` is simply `spago`.
- A `spago` file is included with the package. This is a nodejs script that simply runs an expected sibling `spago.exe` file.
- On installing the package:
  - (On Windows) Yarn/NPM create their shims to call the `spago` script under nodejs. These have no issues.
  - (On Mac/Linux) A symlink is created to the `spago` file as normal.
- After the package is installed, in a `postinstall` script:
  - (On Windows) the `spago.exe` binary file is downloaded, so the nodejs shim will now work.
  - (On Mac/Linux) The binary file is downloaded, which since it's also named `spago`, simply replaces the javascript file, and the symlinks point to this file instead.

Have manually tested on all combinations of the following on Windows: Cmd, Git-Bash, "Linux" (WSL); NPM and Yarn; global and local package installations.

I guess if this patch proves successful then the note about Yarn in the readme can also be removed.

### PR checklist:

- [ ] (Not sure if this is needed?) Updated the version number [in `package.yaml`](https://github.com/spacchetti/spago/blob/master/package.yaml#L2) according to SemVer (`0.x.y.z`, `x` bumps on breaking changes, `y` on additions, `z` on patches)
- [x] Added the change to the "Unreleased" section of the changelog

Fix #187

### CODE REVIEW

**Comment by stkb** [comment]
Don't know why the MacOS test failed. It doesn't even use the npm package?

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Add `spago-curator` tool to generate metadata from the package set (#202)
 
+Bugfixes:
+- Another attempt to fix NPM and Yarn installations on Windows (#187)
```
Comment (f-f) [inline CHANGELOG.md]:
```suggestion
- Another attempt to fix NPM and Yarn installations on Windows (#215, #187)
```

**Comment by stkb** [comment]
OK have added those changes

**Comment by f-f** [comment]
Thank you @stkb!

**Pull request merged**